src/software/pipeline/main_featureExtraction.cpp

```diff
@@ -43,143 +43,149 @@ namespace fs = boost::filesystem;
+        return EXIT_FAILURE;
+    }
+
+    // create output folder
+    if(!fs::exists(outputFolder))
+    {
+        if(!fs::create_directory(outputFolder))
+        {
+            ALICEVISION_LOG_ERROR("Cannot create output folder");
+            return EXIT_FAILURE;
+        }
     }
-  }
 
 #ifdef ALICEVISION_HAVE_GPU_FEATURES
-  // Print GPU Information
-  ALICEVISION_LOG_INFO(gpu::gpuInformationCUDA());
+    // Print GPU Information
+    ALICEVISION_LOG_INFO(gpu::gpuInformationCUDA());
 #endif
 
-  // load input scene
-  sfmData::SfMData sfmData;
-  std::cout << sfmData.getViews().size()  << std::endl;
-  if(!sfmDataIO::Load(sfmData, sfmDataFilename, sfmDataIO::ESfMData(sfmDataIO::VIEWS|sfmDataIO::INTRINSICS)))
-  {
-    ALICEVISION_LOG_ERROR("The input file '" + sfmDataFilename + "' cannot be read");
-    return EXIT_FAILURE;
-  }
-
-  // create feature extractor
-  feature::FeatureExtractor extractor(sfmData);
-  extractor.setMasksFolder(masksFolder);
-  extractor.setOutputFolder(outputFolder);
-
-  // set maxThreads
-  HardwareContext hwc = cmdline.getHardwareContext();
-  hwc.setUserCoresLimit(maxThreads);
-
-  // set extraction range
-  if(rangeStart != -1)
-  {
-    if(rangeStart < 0 || rangeSize < 0 ||
-       rangeStart > sfmData.getViews().size())
+    // load input scene
+    sfmData::SfMData sfmData;
+    std::cout << sfmData.getViews().size()  << std::endl;
+    if(!sfmDataIO::Load(sfmData, sfmDataFilename, sfmDataIO::ESfMData(sfmDataIO::VIEWS|sfmDataIO::INTRINSICS)))
     {
-      ALICEVISION_LOG_ERROR("Range is incorrect");
-      return EXIT_FAILURE;
+        ALICEVISION_LOG_ERROR("The input file '" + sfmDataFilename + "' cannot be read");
+        return EXIT_FAILURE;
     }
 
-    if(rangeStart + rangeSize > sfmData.views.size())
-      rangeSize = sfmData.views.size() - rangeStart;
+    // create feature extractor
+    feature::FeatureExtractor extractor(sfmData);
+    extractor.setMasksFolder(masksFolder);
+    extractor.setOutputFolder(outputFolder);
 
-    extractor.setRange(rangeStart, rangeSize);
-  }
+    // set maxThreads
+    HardwareContext hwc = cmdline.getHardwareContext();
+    hwc.setUserCoresLimit(maxThreads);
 
-  // initialize feature extractor imageDescribers
-  {
-    std::vector<feature::EImageDescriberType> imageDescriberTypes = feature::EImageDescriberType_stringToEnums(describerTypesName);
+    // set extraction range
+    if(rangeStart != -1)
+    {
+        if(rangeStart < 0 || rangeSize < 0)
+        {
+            ALICEVISION_LOG_ERROR("Range is incorrect");
+            return EXIT_FAILURE;
+        }
+
+        if(rangeStart + rangeSize > sfmData.views.size())
+            rangeSize = sfmData.views.size() - rangeStart;
+
+        if(rangeSize < 0)
+        {
+            ALICEVISION_LOG_WARNING("Range size is negative. This can happen if a node's size was smaller than expected.");
+            ALICEVISION_LOG_WARNING("Skipping computations.");
+            return EXIT_SUCCESS;
+        }
```

Comment:
Maybe "<= 0" directly?
```suggestion
        if(rangeSize <= 0)
        {
            ALICEVISION_LOG_WARNING("Nothing to compute.");
            return EXIT_SUCCESS;
        }
```

Reply:
Done.